designs/instance-types-pricing.md

```diff
@@ -0,0 +1,88 @@
+## Design Document for Karpenter: Instance Pricing Override via ConfigMap
```

Comment:
I think we likely want to revive the proposal here to meet the pricing use case, among the others within the same problem space: https://github.com/aws/karpenter-provider-aws/pull/2390.